sympy/matrices/decompositions.py

```diff
@@ -1144,23 +1174,129 @@ def _QRdecomposition(M):
     [14,  21, -14],
     [ 0, 175, -70],
     [ 0,   0,  35]])
+
+    If the matrix is square and full rank, the $Q$ matrix becomes
+    orthogonal in both directions, and needs no augmentation.
+
+    >>> Q * Q.H
+    Matrix([
+    [1, 0, 0],
+    [0, 1, 0],
+    [0, 0, 1]])
+    >>> Q.H * Q
+    Matrix([
+    [1, 0, 0],
+    [0, 1, 0],
+    [0, 0, 1]])
+
     >>> A == Q*R
     True
 
-    QR factorization of an identity matrix:
+    A rank deficient matrix example:
 
-    >>> A = Matrix([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
+    >>> A = Matrix([[12, -51, 0], [6, 167, 0], [-4, 24, 0]])
     >>> Q, R = A.QRdecomposition()
     >>> Q
     Matrix([
+    [ 6/7, -69/175],
+    [ 3/7, 158/175],
+    [-2/7,    6/35]])
+    >>> R
+    Matrix([
+    [14,  21, 0],
+    [ 0, 175, 0]])
+
+    It may return a $Q$ matrix which is orthogonal in only one way
+    around ($\mathbb{I} = Q^H Q$).
+
+    >>> Q.H * Q
+    Matrix([
+    [1, 0],
+    [0, 1]])
+    >>> Q * Q.H
+    Matrix([
+    [27261/30625,   348/30625, -1914/6125],
+    [  348/30625, 30589/30625,   198/6125],
+    [ -1914/6125,    198/6125,   136/1225]])
+
+    If you want to augment the results to be a full orthogonal
+    decomposition, you should augment $Q$ with an another orthogonal
+    column. It may not look easy at the first place.
+
+    But if you know the Gram-Schmidt process, you are able to append an
+    arbitrary standard basis that are linearlly independent to every
```

Comment:
linearly.

I don't think that "if you know" is needed. It's better just to succinctly explain what should be done.